wled00/colors.cpp

```diff
@@ -39,21 +39,20 @@ uint32_t color_add(uint32_t c1, uint32_t c2, bool fast)
 {
   if (c1 == BLACK) return c2;
   if (c2 == BLACK) return c1;
-  if (fast) {
-    uint8_t r = R(c1);
-    uint8_t g = G(c1);
-    uint8_t b = B(c1);
-    uint8_t w = W(c1);
-    r = qadd8(r, R(c2));
-    g = qadd8(g, G(c2));
-    b = qadd8(b, B(c2));
-    w = qadd8(w, W(c2));
+  uint32_t rb = (c1 & 0x00FF00FF) + (c2 & 0x00FF00FF); 
+  uint32_t r = rb >> 16;
+  uint32_t b = rb & 0xFFFF; 
+  uint32_t wg = ((c1>>8) & 0x00FF00FF) + ((c2>>8) & 0x00FF00FF); 
+  uint32_t w = wg >> 16;
+  uint32_t g = wg & 0xFFFF; 
+
+  if (fast) {    
+    r = r > 255 ? 255 : r; 
```

Comment:
Perhaps `r = std:min(255,r)` may be faster?

Reply:
maybe, I have no way of checking. but the compiler is quite clever I found and does optimize.